ui/page/dexclient/markets.go

```diff
@@ -59,116 +72,187 @@ func (pg *Page) ID() string {
 }
 
 func (pg *Page) Layout(gtx C) D {
-	var body func(gtx C) D
-
-	switch {
-	case !pg.WL.MultiWallet.IsConnectedToDecredNetwork():
-		body = func(gtx C) D {
-			return pg.pageSections(gtx, func(gtx C) D {
-				return pg.welcomeLayout(gtx, pg.sync)
-			})
-		}
-	case !pg.Dexc().Initialized():
-		body = func(gtx C) D {
-			return pg.pageSections(gtx, func(gtx C) D {
-				return pg.welcomeLayout(gtx, pg.initialize)
-			})
-		}
-	case !pg.Dexc().IsLoggedIn():
-		body = func(gtx C) D {
-			return pg.pageSections(gtx, func(gtx C) D {
-				return pg.welcomeLayout(gtx, pg.login)
-			})
-		}
-	case len(pg.Dexc().DEXServers()) == 0:
-		body = func(gtx C) D {
-			return pg.pageSections(gtx, func(gtx C) D {
-				return pg.welcomeLayout(gtx, pg.addDex)
-			})
-		}
-	default:
-		body = func(gtx C) D {
-			return pg.pageSections(gtx, pg.registrationStatusLayout)
+	body := func(gtx C) D {
+		switch {
+		case !pg.WL.MultiWallet.IsConnectedToDecredNetwork():
+			return pg.pageSections(gtx, pg.welcomeLayout(pg.syncBtn))
+		case !pg.Dexc().Initialized():
+			return pg.pageSections(gtx, pg.welcomeLayout(pg.initializeBtn))
+		case !pg.Dexc().IsLoggedIn():
+			return pg.pageSections(gtx, pg.welcomeLayout(pg.loginBtn))
+		case pg.dex == nil:
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(pg.headerLayout()),
+				layout.Rigid(func(gtx C) D {
+					return pg.pageSections(gtx, pg.welcomeLayout(pg.addDexBtn))
+				}),
+			)
+		default:
+			if !pg.dex.Connected {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(pg.headerLayout()),
+					layout.Rigid(func(gtx C) D {
+						return pg.pageSections(gtx,
+							pg.Theme.Label(values.TextSize16, fmt.Sprintf("Connection to dex server %s failed. You can close app and try again later or wait for it to reconnect", pg.dex.Host)).Layout)
+					}),
+				)
+			}
+
+			if pg.dex.PendingFee != nil {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(pg.headerLayout()),
+					layout.Rigid(func(gtx C) D {
+						return pg.pageSections(gtx, pg.registrationStatusLayout())
+					}),
+				)
+			}
+
+			pg.initMiniTradeForm()
+
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(pg.headerLayout()),
+				layout.Rigid(func(gtx C) D {
+					return pg.pageSections(gtx, pg.miniTradeFormWdg.layout)
+				}),
+			)
 		}
 	}
 
 	return components.UniformPadding(gtx, body)
 }
 
+func (pg *Page) headerLayout() layout.Widget {
+	return func(gtx C) D {
+		bottom := layout.Inset{Bottom: values.MarginPadding15}
```

Comment:
this is not needed as it was only used once on line 152. It can fit into the same line as it was used. Kindly remove.